src/Tools/LeanCode.ContractsGenerator/CodeGenerator.cs

```diff
@@ -183,7 +183,7 @@ private IEnumerable<InterfaceStatement> GenerateClassesAndInterfaces(SemanticMod
 
             var classes = classesDeclarations
                 .Select(c => model.GetDeclaredSymbol(c))
-                .Union(structsDeclarations.Select(s => model.GetDeclaredSymbol(s)))
```

Comment:
Why?

Reply:
I thought we are looking at declarations here, which should not declare nullable symbols, but I will change this to `SymbolEqualityComparer.IncludeNullability` if you believe it to be more correct in this context.

Reply:
The only [documentation I've found on the subject](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.symbolequalitycomparer.default?view=roslyn-dotnet-3.9.0) claims that `SymbolEqualityComparer.Default` is the one equivalent to calling `IEquatable<T>.Equals(T)` and not the other one.

Reply:
Ugh, so they changed it once again. There was a time where official analyzers suggested using it explicitly. Thanks for the explanation.

Nevertheless, it doesn't really matter here - this generator will be hopefully removed soon.

Reply:
Ugh, so they changed it once again. There was a time where official analyzers suggested using it explicitly. Thanks for the explanation.

Nevertheless, it doesn't really matter here - this generator will be hopefully removed soon.

Reply:
> There was a time where official analyzers suggested using it explicitly.

The analyzer suggesting using a comparator explicitly was exactly the reason why I've changed that, it just doesn't force you to use any particular one, only that you shouldn't rely on implicit default that comes from implementation of `IEquatable<T>`.

> Nevertheless, it doesn't really matter here - this generator will be hopefully removed soon.

:+1: